llvm/lib/CodeGen/AsmPrinter/AccelTable.cpp

```diff
@@ -395,36 +401,92 @@ void Dwarf5AccelTableWriter::Header::emit(Dwarf5AccelTableWriter &Ctx) {
   Asm->OutStreamer->emitBytes({AugmentationString, AugmentationStringSize});
 }
 
-static uint32_t constexpr LowerBitSize = dwarf::DW_IDX_type_hash;
+std::optional<uint64_t>
+DWARF5AccelTableData::getDefiningParentDieOffset(const DIE &Die) {
+  auto *Parent = Die.getParent();
+  if (!Parent)
+    return {};
+  if (Parent->findAttribute(dwarf::Attribute::DW_AT_declaration))
+    return {};
+  return Parent->getOffset();
```

Comment:
Why not
```
if (auto *Parent = Die.getParent()) 
  if (Parent->findAttribute(dwarf::Attribute::DW_AT_declaration))
    return Parent->getOffset();
return {};
```

Reply:
The logic is flipped in the suggestion -- it should be `!Parent->find...` -- but it still looks better, so I will change it!